lib/ansible/modules/tacp/tacp_instance.py

```diff
@@ -339,18 +560,35 @@ def instance_power_action(name, api_client, action):
         else:
             # Application does not exist yet, so create it
             instance_params = generate_instance_params(module)
-            create_instance(instance_params, api_client)
+
+            # Create instance with disks + vnics from the template ONLY
+            created_instance_uuid = create_instance(
+                instance_params, api_client).object_uuid
             current_state = State.SHUTDOWN
 
+            # Add vnics + disks from playbook specification
+            # and modify boot order to match
+            template_resource = tacp_utils.TemplateResource(api_client)
+            template_results = template_resource.filter(
+                name=("==", module.params['template']))
+
+            if template_results:
+                template_dict = template_results[0].to_dict()
+            else:
+                fail_with_reason("Template {} is not present.".format(
+                    module.params['template']))
+
+            boot_order = get_boot_order(module, template_dict)
+
+            if len(boot_order) > len(template_dict['boot_order']):
+                add_vnics_and_disks_and_update_boot_order(module,
```

Comment:
You dont really need to indent it that much. Other example

```suggestion
                add_vnics_and_disks_and_update_boot_order(
                    module, created_instance_uuid, boot_order
                )
```